libs/core/langchain_core/runnables/history.py

```diff
@@ -241,7 +241,11 @@ def __init__(
         self,
         runnable: Union[
             Runnable[
-                Union[MessagesOrDictWithMessages],
+                list[BaseMessage],
```

Comment:
I think this supported Sequence (which is both tuple and list)

Reply:
https://github.com/langchain-ai/langchain/blob/29e17fbd6bcba110b4004a3bbd4697fb3cc400eb/libs/core/langchain_core/runnables/history.py#L37-L37

Reply:
Counter-intuitively, this would not be nice to users as it means their runnable must be able to handle a `Sequence` whereas RunnableWithMessageHistory always calls it with a `list`. So for the open-close principle, using `list` is better than `Sequence` here.